imports/wasi_snapshot_preview1/poll.go

```diff
@@ -166,7 +166,7 @@ func pollOneoffFn(_ context.Context, mod api.Module, params []uint64) sys.Errno
 		// We only need to observe the timeout (nonzero if there are clock subscriptions)
 		// and return.
 		if timeout > 0 {
-			sysCtx.Nanosleep(int64(timeout))
+			sysCtx.Nanosleep(ctx, int64(timeout))
```

Comment:
Is this really enough? This is not keeping the typical promise of sleep: sleeping _at least_ `n`ns.

Are we sure wazero immediately errors out on the context after this? Or is it possible for guest code to continue assuming poll just timed out?

Reply:
I should probably add an integration test